frontend/src/utils/handleCaptionsSignal/handleCaptionsSignal.ts

```diff
@@ -0,0 +1,156 @@
+ */
+const handleCaptionsSignal = ({
+  event,
+  currentCaptionsIdRef,
+  captionsActiveCountRef,
+  currentRoomNameRef,
+  vonageVideoClient,
+  setCaptionsEnabled,
+}: CaptionsHandleType) => {
+  try {
+    const parsedData: CaptionsSignalDataType = JSON.parse(event.data as string);
+    const { action, captionsId, currentCount } = parsedData;
+
+    switch (action) {
+      case 'enable':
+        if (captionsId) {
+          currentCaptionsIdRef.current = captionsId;
+          setCaptionsEnabled(true);
+          vonageVideoClient?.signal({
+            type: 'captions',
+            data: JSON.stringify({
+              action: 'update-current-user-count',
+              currentCount: captionsActiveCountRef.current + 1,
+            }),
+          });
+        }
+        break;
+
+      case 'join':
+        setCaptionsEnabled(true);
+        captionsActiveCountRef.current += 1;
+        break;
+
+      case 'update-current-user-count':
+        if (currentCount) {
+          captionsActiveCountRef.current = currentCount;
+        }
+        break;
+
+      case 'leave': {
+        const newCount = Math.max(0, captionsActiveCountRef.current - 1);
+        setCaptionsEnabled(false);
+
+        // If there are no other participants using captions, we disable them for the whole session.
+        // This is to ensure that captions are only disabled when there are other participants using them.
+        if (newCount === 0 && currentCaptionsIdRef.current && currentRoomNameRef.current) {
+          disableCaptions(currentRoomNameRef.current, currentCaptionsIdRef.current)
+            .then(() => {
+              currentCaptionsIdRef.current = null;
+
+              vonageVideoClient?.signal({
+                type: 'captions',
+                data: JSON.stringify({ action: 'disable' }),
+              });
+            })
+            .catch((err) => console.error('Error disabling captions:', err));
+        }
+
+        // If there are participants remaining, we signal the new count to other participants so they can update their tracking of the captions
+        if (newCount > 0) {
+          vonageVideoClient?.signal({
+            type: 'captions',
+            data: JSON.stringify({
+              action: 'update-current-user-count',
+              currentCount: newCount,
+            }),
+          });
+        }
+        break;
+      }
+
+      case 'disable':
+        // We turned off the captions session-wide
+        currentCaptionsIdRef.current = null;
+        captionsActiveCountRef.current = 0;
+        break;
+
+      // Handle the case of sending out captions ID when requested
+      case 'request-status':
+        vonageVideoClient?.signal({
+          type: 'captions',
+          data: JSON.stringify({
+            action: 'status-response',
+            captionsId: currentCaptionsIdRef.current,
+            currentCount: captionsActiveCountRef.current,
+          }),
+        });
```

Comment:
Instead of handling signaling through clients, can we move signaling to the backend?